ddtrace/internal/codeowners.py

```diff
@@ -119,16 +118,20 @@ class Codeowners(object):
         ".gitlab/CODEOWNERS",
     )
 
-    def __init__(self, path=None, cwd=None):
-        # type: (Optional[str], Optional[str]) -> None
+    def __init__(self, path: Optional[str] = None, cwd: Optional[str] = None):
         """Initialize Codeowners object.
 
         :param path: path to CODEOWNERS file otherwise try to use any from known locations
         """
+        self.patterns: List[Tuple[re.Pattern, List[str]]] = []
+        self.path: Optional[str] = None
+
         path = path or self.location(cwd)
-        if path is not None:
-            self.path = path  # type: str
-        self.patterns = []  # type: List[Tuple[re.Pattern, List[str]]]
+
+        if path is None:
+            raise ValueError("CODEOWNERS file not found")
+
+        self.path = path
```

Comment:
Execution only reaches this point if `path` is not `None`, right? So why is `self.path` optional?

Reply:
I just wanted it to have a default value and so that `mypy` wouldn't complain when [I check for `None` in parse()](https://github.com/DataDog/dd-trace-py/pull/10423/files#diff-68384ccfde4c133ff7fb33b050476b2fa3efeac1ced057790ef2fe03df140777R148).

It's an abundance of caution for something impossible.